### PR TITLE
Migration to 6x6 APE matrix in alignment and RECO

### PR DESCRIPTION
**Pull request by ASvyatkovskiy**
The pull request implements changes in a set of classes, plugins and configuration files necessary
for migration to the 6x6 APE matrix in the detector alignment and muon reconstruction parts of the CMSSW.
It affects the Alignment, Geometry, DataFormats and RecoMuon subsystems of CMSSW.

The previous relevant pull request is: https://github.com/cms-sw/cmssw/pull/6420
Large fraction of the changes in this PR are simply due to migration to the "extended" CondFormat.

Note: the new Global Tag with extended payloads is necessary for the PR to work, which is in "Pending" state at the moment

### CODE REVIEW

File: Geometry/CommonDetUnit/interface/MuonGeomDet.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#ifndef CommonDet_MuonGeomDet_H
```
Comment (ianna) [inline Geometry/CommonDetUnit/interface/MuonGeomDet.h]: This file does not seem to be relevant to the proposed migration.

File: Geometry/Records/interface/CaloTowerGeometryRecord.h [hunk 2]
```diff
@@ -27,6 +28,7 @@ class CaloTowerGeometryRecord :
                 IdealGeometryRecord,
 		CaloTowerAlignmentRcd, 
 		CaloTowerAlignmentErrorRcd,
```
Comment (ianna) [inline Geometry/Records/interface/CaloTowerGeometryRecord.h]: Can this dependency on *AlignementErrorRcd be moved to the *AlignementErrorExtendedRcd?

File: Geometry/CommonDetUnit/interface/MuonGeomDet.h (hunk 1 above)
Comment (ianna) [inline Geometry/CommonDetUnit/interface/MuonGeomDet.h]: ok, thanks. I'd think, this class should be a part of a separate PR for MuonGeomDet migration.

**Comment by ktf** [comment]
Do you actually want to go forward with this PR? We should close otherwise.

File: Alignment/OfflineValidation/test/eopTreeWriter_cfg.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
 # load configuration files
 process.load("Configuration.StandardSequences.MagneticField_cff")
-process.load("Configuration.Geometry.GeometryDB_cff")
+process.load("Configuration.StandardSequences.Geometry_cff")
```
Comment (ianna) [inline Alignment/OfflineValidation/test/eopTreeWriter_cfg.py]: This change of include will give you a wrong geometry. I think, you need to use:
```
process.load("Configuration.Geometry.GeometryRecoDB_cff")
```
BTW, this comment is valid for all other similar changes.

**Comment by slava77** [comment]
@diguida @rcastello @apfeiffer1 
what is the status of the global tag?
I can't figure out if things actually went ok by looking at the dropbox
https://cms-conddb.cern.ch/logs/dropBox/searchUserLog?beginDate=2014-12-02&fileName=APE&endDate=2015-01-09&metadata=&fileHash=&userText=&username=&backend=&statusCode=Any

File: Alignment/OfflineValidation/test/eopTreeWriter_cfg.py (hunk 3 above)
Comment (ASvyatkovskiy) [inline Alignment/OfflineValidation/test/eopTreeWriter_cfg.py]: @ianna thanks for pointing that out - I will fix that.

Comment (ASvyatkovskiy) [inline Alignment/OfflineValidation/test/eopTreeWriter_cfg.py]: Hi @ianna, going back to this: I actually did not explicitly make this change, it must have happened as a result of an update which happened after I prepared the branch but before I created a PR (?).
I checked, and actually both "Configuration.Geometry.GeometryDB_cff" and 
"Configuration.StandardSequences.Geometry_cff" appear in the files. Namely:

bash-4.1$ while read file;do find $file -name "*" | xargs grep -l Configuration.Geometry.GeometryDB_cff; done < list_to_change
Alignment/OfflineValidation/python/TkAlAllInOneTool/alternateValidationTemplates.py
Alignment/OfflineValidation/python/TkAlAllInOneTool/geometryComparisonTemplates.py
Alignment/OfflineValidation/test/PVValidation_TEMPL_cfg.py
Alignment/OfflineValidation/test/dqmOfflinevalidator_cfg.py
Alignment/OfflineValidation/test/offlinevalidator_cfg.py
Alignment/OfflineValidation/test/testIntoNtuples_cfg.py
Alignment/OfflineValidation/test/testSplitterValidation_cfg.py

bash-4.1$ while read file;do find $file -name "*" | xargs grep -l Configuration.StandardSequences.Geometry_cff; done < list_to_change
Alignment/CommonAlignmentProducer/test/GlobalTrackerMuonAlignment_test_cfg.py
Alignment/MillePedeAlignmentAlgorithm/test/alignment_cfg.py
Alignment/MillePedeAlignmentAlgorithm/test/alignment_forGeomComp_cfg.py
Alignment/MillePedeAlignmentAlgorithm/test/alignment_survey_cfg.py
Alignment/MuonAlignmentAlgorithms/test/runDTLocalMillepede.py
Alignment/MuonAlignmentAlgorithms/test/runDTLocalMillepedeData.py
Alignment/MuonAlignmentAlgorithms/test/runDTLocalMillepedeFit.py
Alignment/OfflineValidation/python/TkAlAllInOneTool/alternateValidationTemplates.py
Alignment/OfflineValidation/test/eopTreeWriter_cfg.py

 Should I change both to what you suggest: "Configuration.Geometry.GeometryRecoDB_cff"? What would be the right thing to do

File: Geometry/Records/interface/CaloTowerGeometryRecord.h (hunk 2 above)
Comment (ASvyatkovskiy) [inline Geometry/Records/interface/CaloTowerGeometryRecord.h]: @ianna to me, *AlignmentErrorRcd should stop existing after everything is tested and validated. And in principle the extended format should become main. 

Otherwise, if you insists I can move the *AlignmentErrorRcd includes to the corresponding*AlignementErrorExtendedRcd.

**Comment by slava77** [comment]
@diguida @rcastello @apfeiffer1  
ping
was there any progress with the GT here?

File: Geometry/Records/interface/CaloTowerGeometryRecord.h (hunk 2 above)
Comment (ianna) [inline Geometry/Records/interface/CaloTowerGeometryRecord.h]: @ASvyatkovskiy - I think, you should remove AlignmentErrorRcd from here then. Otherwise, we'll have to keep it in a GT.

File: Alignment/OfflineValidation/test/eopTreeWriter_cfg.py (hunk 3 above)
Comment (ianna) [inline Alignment/OfflineValidation/test/eopTreeWriter_cfg.py]: @ASvyatkovskiy - yes, using DB or RecoDB (for GTs which do not need/have Sim geometry) would be a right thing to do. Thanks.

**Comment by diguida** [comment]
@slava77 the new set of Global Tags are ready #7339 
@ASvyatkovskiy Either you or I can prepare a new PR with a rebase of this PR on top of the latest IB `CMSSW_7_4_X_2015-01-23-0200` and cherry pick #7339 Let me know.

File: Geometry/Records/interface/CaloTowerGeometryRecord.h (hunk 2 above)
Comment (diguida) [inline Geometry/Records/interface/CaloTowerGeometryRecord.h]: @ASvyatkovskiy I agree with @ianna comment: given that we want to get rid of the old records in production, we should remove the old APE records in all the dependent ones, otherwise the job will crash as the dependent record is missing one of its constituents in the `EventSetup`. For the moment it is ok, as the new GT contains both old and new, but in the near future, when the old ones will be decommissioned, you should remove them.

**Comment by diguida** [comment]
@ASvyatkovskiy #7352 points to a branch with a rebase of this PR on top of the latest IB CMSSW_7_4_X_2015-01-23-0200 + #7339 
I am asking the bot to test. Please run also your tests there.

**Comment by davidlange6** [comment]
I'll close this pr as @diguida just made a more complete one
